From 36b2bc16d44e3481c539c7c696b522b1212b57d8 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:35 -0400
Subject: [usb] hub_activate kill an 'else'

Message-id: <1410887463-20674-84-git-send-email-dzickus@redhat.com>
Patchwork-id: 95285
O-Subject: [RHEL7 PATCH 083/591] usb: hub_activate kill an 'else'
Bugzilla: 1032371
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1032371

commit fd1ac4cf524018198373c5eb90ca0e7968c5bbb8
Author: Dan Williams <dan.j.williams@intel.com>
Date:   Mon Oct 7 11:58:20 2013 -0700

    usb: hub_activate kill an 'else'

    Remove a few extra lines and make it clear that all implementations
    disable the port by sharing the same line of code.

    Acked-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Dan Williams <dan.j.williams@intel.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index 1733ba4..bd6986e 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -1112,16 +1112,13 @@ static void hub_activate(struct usb_hub *hub, enum hub_activation_type type)
    /*
     * USB3 protocol ports will automatically transition
     * to Enabled state when detect an USB3.0 device attach.
-    * Do not disable USB3 protocol ports.
+    * Do not disable USB3 protocol ports, just pretend
+    * power was lost
     */
-   if (!hub_is_superspeed(hdev)) {
+   portstatus &= ~USB_PORT_STAT_ENABLE;
+   if (!hub_is_superspeed(hdev))
     usb_clear_port_feature(hdev, port1,
          USB_PORT_FEAT_ENABLE);
-    portstatus &= ~USB_PORT_STAT_ENABLE;
-   } else {
-    /* Pretend that power was lost for USB3 devs */
-    portstatus &= ~USB_PORT_STAT_ENABLE;
-   }
   }
 
   /* Clear status-change flags; we'll debounce later */
-- 
1.7.1